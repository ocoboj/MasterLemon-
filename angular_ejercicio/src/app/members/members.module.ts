import { NgModule } from '@angular/core';
import { FormsModule } from '@angular/forms';
import { HttpClientModule } from '@angular/common/http'
import { CommonModule } from '@angular/common';
import {
  MemberRowComponent,
  MemberHeadComponent,
  MembersTableComponent
} from './members-table';

@NgModule({
  imports: [
    CommonModule,
    HttpClientModule,
    FormsModule
  ],
  declarations: [
    MemberRowComponent,
    MemberHeadComponent,
    MembersTableComponent,
  ],
  exports: [
    MembersTableComponent
  ]
})
export class MembersModule { }
