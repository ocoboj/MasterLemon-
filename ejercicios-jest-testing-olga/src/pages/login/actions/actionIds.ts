export const actionIds = {
  UPDATE_LOGIN_ENTITY_FIELD: 'UPDATE_LOGIN_ENTITY_FIELD',
  UPDATE_LOGIN_FORM_ERRORS: 'UPDATE_LOGIN_FORM_ERRORS',
};
