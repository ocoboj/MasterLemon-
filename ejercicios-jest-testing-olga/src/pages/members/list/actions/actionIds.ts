export const actionIds = {
  UPDATE_MEMBERS: 'UPDATE_MEMBERS',
};
