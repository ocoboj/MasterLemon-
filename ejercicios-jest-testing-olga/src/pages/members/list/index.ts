export * from './pageContainer';
export * from './reducers';
