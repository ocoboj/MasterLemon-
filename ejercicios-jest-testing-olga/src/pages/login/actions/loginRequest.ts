import { FieldValidationResult } from 'lc-form-validation';
import { validations } from '../validations';
import { LoginEntity, createEmptyLoginFormErrors } from '../viewModel';
import { mapLoginEntityVMToModel } from '../mappers';
import { login } from '../../../rest-api/api/login';
import { history } from '../../../history';
import { routes } from '../../../common/constants/routes';
import { actionIds } from './actionIds';

export const loginRequest = (loginEntity: LoginEntity) => (dispatch) => (
  validations.validateForm(loginEntity)
    .then((formValidationResult) => {
      formValidationResult.succeeded ?
        doLogin(loginEntity) :
        dispatch(updateLoginFormErrors(formValidationResult.fieldErrors));
    })
);

const doLogin = (loginEntity: LoginEntity) => {
  const loginEntityModel = mapLoginEntityVMToModel(loginEntity);
  login(loginEntityModel)
    .then(() => {
      history.push(routes.members);
    })
    .catch(console.log);
};

const updateLoginFormErrors = (fieldErrors: { [key: string]: FieldValidationResult }) => ({
  type: actionIds.UPDATE_LOGIN_FORM_ERRORS,
  payload: fieldErrors,
});
