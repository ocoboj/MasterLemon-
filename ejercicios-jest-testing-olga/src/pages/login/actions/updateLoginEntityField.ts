import { FieldValidationResult } from 'lc-form-validation';
import { actionIds } from './actionIds';
import { validations } from '../validations';
import { LoginEntity } from '../viewModel';

export const updateLoginEntityField = (loginEntity: LoginEntity, fieldName: string, value: any) => (dispatch) =>
  validations.validateField(loginEntity, fieldName, value)
    .then((fieldValidationResult) => {
      dispatch(updateLoginEntityFieldCompleted(fieldName, value, fieldValidationResult));
    });

export const updateLoginEntityFieldCompleted =
  (fieldName: string, value: any, fieldValidationResult: FieldValidationResult) => ({
    type: actionIds.UPDATE_LOGIN_ENTITY_FIELD,
    payload: {
      fieldName,
      value,
      fieldValidationResult,
    },
  });
