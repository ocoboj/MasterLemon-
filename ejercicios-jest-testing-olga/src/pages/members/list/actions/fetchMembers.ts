import { actionIds } from './actionIds';
import { Member } from '../../../../rest-api/model';
import * as apiMember from '../../../../rest-api/api/member';

export const fetchMembers = () => (dispatch) => (
  apiMember.fetchMembers()
    .then((members) => dispatch(fetchMembersCompleted(members)))
    .catch(console.log)
);

const fetchMembersCompleted = (members: Member[]) => ({
  type: actionIds.UPDATE_MEMBERS,
  payload: members,
});
