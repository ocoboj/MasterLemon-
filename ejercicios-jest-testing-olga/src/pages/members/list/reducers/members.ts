import { actionIds } from '../actions/actionIds';
import { Member } from '../../../../rest-api/model';

export type MembersState = Member[];

export const membersReducer = (state = [], action): MembersState => {
  switch (action.type) {
    case actionIds.UPDATE_MEMBERS:
      return handleUpdateMembers(state, action.payload);
  }

  return state;
};

const handleUpdateMembers = (state: MembersState, members: Member[]): MembersState => (
  members
);
