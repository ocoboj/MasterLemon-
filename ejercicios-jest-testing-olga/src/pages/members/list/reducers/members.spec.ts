import { MembersState, membersReducer } from "./members";
import deepFreeze from 'deep-freeze';
import { actionIds } from "../actions/actionIds";

describe('pages/members/list/reducers/members spec', () => {
    it('should return same state when passing a state and some action type', () => {
        
        const state: MembersState = [
            {
                id: 5,
                login: 'test login 5',
                avatar_url: 'test avatar_url 5'
            }
        ];

        const action = {
            type: 'some type'
        };

        deepFreeze(state);

        const nexState = membersReducer(state, action);

        expect(nexState).toEqual([
            {
                id: 5,
                login: 'test login 5',
                avatar_url: 'test avatar_url 5'
            }
        ]);
        expect(nexState).toMatchSnapshot();
    });

    it('should return update when passing state, members payload and UPDATE_MEMBERS', () => {
        const state: MembersState = [
            {
                id: 3,
                login: 'test login 3',
                avatar_url: 'test avatar_url 3',
                
            }
        ];

        const payload = [
            {
                id: 9,
                login: 'test login 9',
                avatar_url: 'test avatar_url 9'
            }
        ]

        const action = {
            type: actionIds.UPDATE_MEMBERS,
            payload,
        };

        deepFreeze(state)

        const nexState = membersReducer(state, action);

        expect(nexState).toEqual(payload);
        expect(nexState).toMatchSnapshot();
    });
})