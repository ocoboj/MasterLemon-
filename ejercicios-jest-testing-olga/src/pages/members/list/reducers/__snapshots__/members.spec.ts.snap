// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages/members/list/reducers/members spec should return same state when passing a state and some action type 1`] = `
Array [
  Object {
    "avatar_url": "test avatar_url 5",
    "id": 5,
    "login": "test login 5",
  },
]
`;

exports[`pages/members/list/reducers/members spec should return update when passing state, members payload and UPDATE_MEMBERS 1`] = `
Array [
  Object {
    "avatar_url": "test avatar_url 9",
    "id": 9,
    "login": "test login 9",
  },
]
`;
